### PR TITLE
Update namespace used in Digest::UUID

### PR DESCRIPTION
**Pull request by ryan-mcneil**
## Summary

- In preparation for bumping rails to 7.1+, we need to update how we're generating UUIDs per a new [standard](https://www.ietf.org/rfc/rfc4122.txt) that's since enforced by ActiveSupport 7.1.0
- We're required to use deterministic UUIDs in these cases so that their uniqueness allows us to use them for redis keys, debugging in logs, etc and the ability to reproduce the same UUID allows us to maintain consistency between sessions
- We can no longer provide a string, but must include a UUID as the first arg to the `Digest::UUID` methods, so we've decided to just hardcode randomly generated UUIDs into constants so that we can continue to use this strategy and have access to the variable if needed for debugging, etc.
- The UUID used across factories/specs was originally `3097e489-ad75-5746-ab1a-e0aabc1b426a` which is the output of `IAMUserIdentity#uuid`  using the old definition (`Digest::UUID.uuid_v5(@iam_sec_id, @icn)`). I've updated this value across the repo to `885b29b2-87bf-5791-b7ca-2bb86cf3d6f5` which is the output given the new definition (`Digest::UUID.uuid_v5(IAM_NAMESPACE, @icn)`)

NOTE: the following message from the spec logs will be removed in 7.1 once we fix this namespace issue

```
DEPRECATION WARNING: Providing a namespace ID that is not one of the constants defined on Digest::UUID generates an incorrect UUID value according to RFC 4122. To enable the correct behavior, set the Rails.application.config.active_support.use_rfc4122_namespaced_uuids configuration option to true. (called from uuid at /srv/vets-api/src/app/models/iam_user_identity.rb:75)
```

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/73307

## Testing done

- [x] *New code is covered by unit tests*

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Review by kpethtel — DISMISSED** [review]
LGTM.

File: lib/iam_ssoe_oauth/session_manager.rb [hunk 1]
```diff
@@ -12,6 +12,7 @@ def initialize(access_token)
     end
 
     def find_or_create_user
+      # require 'pry'; binding.pry;
```
Comment (bosawt) [inline lib/iam_ssoe_oauth/session_manager.rb:15]: can you remove this?

File: app/models/iam_user_identity.rb [hunk 2]
```diff
@@ -70,7 +72,8 @@ def set_expire
   # @return [String] UUID that is unique to this user
   #
   def uuid
-    Digest::UUID.uuid_v5(@iam_sec_id, @icn)
+    Digest::UUID.uuid_v5(IAM_NAMESPACE, @icn)
+    # Digest::UUID.uuid_v5(@iam_sec_id, @icn)
```
Comment (bosawt) [inline app/models/iam_user_identity.rb:76]: We prob don't need this comment, we can just have users look up the git history for the history here I'd say

**Comment by ryan-mcneil** [comment]
@bosawt @kpethtel yeah, sorry, still a draft. removing those lines broke a handful of specs. I'm trying to figure out the problem now. I'll clean up the prys/comments, I just had to quickly switch gears before.

File: app/models/iam_user_identity.rb (hunk 2 above)
Comment (ryan-mcneil) [reply app/models/iam_user_identity.rb:76]: yeah it was just there to toggle between the two while i'm debugging the spec failures. It'll be removed

File: modules/mobile/spec/support/fixtures/claims_and_appeals.json [hunk 3]
```diff
@@ -5,7 +5,7 @@
     "data": {},
     "created_at": "2020-04-05T18:35:16.414Z",
     "updated_at": "2020-04-05T18:35:16.414Z",
-    "user_uuid": "3097e489-ad75-5746-ab1a-e0aabc1b426a",
+    "user_uuid": "885b29b2-87bf-5791-b7ca-2bb86cf3d6f5",
```
Comment (ryan-mcneil) [inline modules/mobile/spec/support/fixtures/claims_and_appeals.json:8]: from bullet 4 in the definition, I replaced all references of the old test data UUID with the new one

**Comment by ryan-mcneil** [comment]
FYI I think the specs are failing because of a bad/flaky test (`/spec/sidekiq/central_mail/submit_form4142_job_spec.rb:58`) that's failing because of timezones. I'll re-run in the morning. My guess is the test passes before ~7pm ET 😆

**Review by bosawt — APPROVED** [review]
theoretically makes sense. Unfortunately I'm not able to test since this impacts deprecated mobile auth but the changes seem sensible (also code ownership of IAM user identity probably should move to mobile but overall it doesn't really matter since I believe we're trying to remove it in general soon)

**Review by kpethtel — APPROVED** [review]
LGTM. Thanks for doing this.

**Pull request merged**